snuba/utils/streams/consumer.py

```diff
@@ -70,6 +70,33 @@ def subscribe(
     def poll(self, timeout: Optional[float] = None) -> Optional[Message[TPayload]]:
         raise NotImplementedError
 
+    @abstractmethod
+    def tell(self) -> Mapping[Partition, int]:
+        """
+        Return the working offsets for all currently assigned positions.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def seek(self, offsets: Mapping[Partition, int]) -> None:
+        """
+        Update the working offsets for the provided partitions.
+
+        When using this method, it is possible to set a partition to an
+        invalid offset without an immediate error. (Examples of invalid
+        offsets include an offset that is too low and has already been
+        dropped by the broker due to data retention policies, or an offset
+        that is too high which is not yet associated with a message.) Since
+        this method only updates the local working offset (and does not
+        communicate with the broker), setting an invalid offset will cause a
+        subsequent ``poll`` call to raise an exception, even though the call
+        to ``seek`` succeeded.
+
+        If any provided partitions are not in the assignment set, an
+        exception will be raised and no offsets will be modified.
```

Comment:
It probably would make sense to make this a specific subtype of `ConsumerError` for testing purposes, I guess.